lib/chef/provider/git.rb

```diff
@@ -167,12 +167,10 @@ def enable_submodules
         if @new_resource.enable_submodules
           converge_by("enable git submodules for #{@new_resource}") do
             Chef::Log.info "#{@new_resource} synchronizing git submodules"
-            command = "git submodule sync"
-            shell_out!(command, run_options(:cwd => @new_resource.destination))
+            git(%w{submodule sync}, { cwd: cwd })
```

Comment:
I have this updated (and it look much cleaner).  I'll be pushing an update shortly.